src/main/java/mezz/jei/util/ErrorUtil.java

```diff
@@ -221,6 +222,32 @@ public static void checkNotEmpty(@Nullable Collection<?> values, String name) {
 		}
 	}
 
+	public static void checkNotEmpty(@Nullable FluidStack fluidStack, String name) {
+		if (fluidStack == null) {
+			throw new NullPointerException(name + " must not be null.");
+		} else if (fluidStack.isEmpty()) {
```

Comment:
For ItemStack, we check if it's empty because empty ones contain "air". Do empty fluidstacks have similar behavior? If not, we probably just want to check that this is not null.

Reply:
Perhaps sticking my nose where it doesn't belong, but I'm concerned by the FluidStack null check compared to the previous isEmpty check.

There IS a Fluid version of "Air" (net.minecraft.fluid.EmptyFluid), which a FluidStack will return if isEmpty is true.  That is, a FluidStack, like an ItemStack, shouldn't be null.

Of course, I don't know how this all plays out inside JEI internally, so I'm yipping up an inappropriate tree here, I'll just slink away quietly...